server/api.go

```diff
@@ -595,3 +599,56 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
 
 	p.writeAPIResponse(w, info.Settings)
 }
+
+type SubscriptionResponse struct {
+	RepositoryName string   `json:"repository_name"`
+	RepositoryURL  string   `json:"repository_url"`
+	Features       []string `json:"features"`
+	CreatorID      string   `json:"creator_id"`
+}
+
+func subscriptionsToResponse(config *configuration, subscriptions []*subscription.Subscription) []SubscriptionResponse {
+	gitlabURL, _ := url.Parse(config.GitlabURL)
+
+	subscriptionResponses := make([]SubscriptionResponse, 0, len(subscriptions))
+
+	for _, subscription := range subscriptions {
+		features := []string{}
+		if len(subscription.Features) > 0 {
+			features = strings.Split(subscription.Features, ",")
+		}
+
+		subscriptionResponses = append(subscriptionResponses, SubscriptionResponse{
+			RepositoryName: subscription.Repository,
+			RepositoryURL:  gitlabURL.JoinPath(subscription.Repository).String(),
```

Comment:
I just ran into this while deploying:

`JoinPath` was added in Go 1.19 https://pkg.go.dev/net/url#URL.JoinPath

The go.mod for this project states that it works with go 1.16.

The mattermost-server project requires go 1.18.

Reply:
Thank you for this: I didn't even check the supported versions on this method!